algebra/sfem_LpSmoother.hpp

```diff
@@ -0,0 +1,45 @@
+#ifndef SFEM_LP_SMOOTHER_HPP
+#define SFEM_LP_SMOOTHER_HPP
+
+#include <cstddef>
+#include <memory>
+
+#include "sfem_Buffer.hpp"
+#include "sfem_MatrixFreeLinearSolver.hpp"
+#include "sfem_openmp_blas.hpp"
+
+namespace sfem {
+    template <typename T>
+    class LpSmoother final : public Operator<T> {
+    public:
+        ExecutionSpace execution_space_{EXECUTION_SPACE_INVALID};
+        ptrdiff_t n_dofs{-1};
+        BLAS_Tpl<T> blas;
+        std::shared_ptr<Buffer<T>> inv_diag;
+
+        void default_init() {
+            OpenMP_BLAS<T>::build_blas(blas);
+            execution_space_ = EXECUTION_SPACE_HOST;
+        }
+
+        /* Operator */
+        int apply(const T* const b, T* const x) override {
+            blas.axpyz(n_dofs, inv_diag, b, 1, x);
```

Comment:
Ah I guess I should have used alpha = 0 here

Reply:
Actually I always allow Operators to add, so this is fine

Reply:
In fact I set the output vector to zero if I want to make sure there is no add outside

Reply:
but it does feel strange to expect the preconditioner to apply the correction so maybe this inconsistency is not good. That's how I left it in the new commit though